<a href="https://colab.research.google.com/github/bjkc/Data-visualization/blob/main/review_class.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**bold text**


## Review todays **classwork**

Data use is **bigquery-public-data.london_fire_brigade. fire_brigade_service_calls**

In [ ]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


This code will is used to authorize the permission to use the data which included several process. 

In [ ]:
%%bigquery --project data-visualization-bjkc
SELECT 
  COUNT(*) as total_rows
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`

total_rows
0       32247

It looks like there is total of 32247 rows in the data set which is a lot. Lets have a look at the data and find out the attributes.

In [ ]:
%%bigquery --project data-visualization-bjkc
SELECT *
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`
LIMIT 15

incident_number  ... num_pumps_attending
0   000320-01012017  ...                   2
1   003727-09012017  ...                   2
2   003762-09012017  ...                   2
3   006403-16012017  ...                   2
4   010947-26012017  ...                   1
5   014169-03022017  ...                   5
6   016263-08022017  ...                   2
7   016865-09022017  ...                   1
8   017092-10022017  ...                   1
9   024609-27022017  ...                   1
10  025352-01032017  ...                   1
11  025355-01032017  ...                   2
12  033935-21032017  ...                   2
13  034332-22032017  ...                   2
14  034347-22032017  ...                   2

[15 rows x 32 columns]

This input gives me only 15 rows even though it has total of 32247 rows. That is really helpful to over view in a small table which beautifully presents the information. 

In [ ]:
%%bigquery --project data-visualization-bjkc
SELECT Count(*) as number_of_late_night_false_alarm
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`
WHERE incident_group = "False Alarm" AND (hour_of_call >= 22 OR hour_of_call <= 5)

number_of_late_night_false_alarm
0                              3098

In [ ]:
%%bigquery --project data-visualization-bjkc
SELECT incident_group, AVG(num_pumps_attending) as avg_pumps_attending
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`
GROUP BY incident_group

incident_group  avg_pumps_attending
0      False Alarm             1.725776
1             Fire             1.633012
2  Special Service             1.177192